crates/core/tedge_agent/src/tedge_to_te_converter/tests.rs

```diff
@@ -85,10 +85,15 @@ async fn convert_incoming_main_device_alarm_topic() -> Result<(), DynError> {
 }
 
 fn same_json_over_mqtt_msg(left: &MqttMessage, right: &MqttMessage) -> bool {
-    let left_msg: serde_json::Value = serde_json::from_slice(left.payload.as_bytes()).unwrap();
-    let right_msg: serde_json::Value = serde_json::from_slice(right.payload.as_bytes()).unwrap();
-
-    (left.topic == right.topic) && (left_msg == right_msg)
+    if !left.payload_str().unwrap().is_empty() && !right.payload_str().unwrap().is_empty() {
+        let left_msg: serde_json::Value = serde_json::from_slice(left.payload.as_bytes()).unwrap();
+        let right_msg: serde_json::Value =
+            serde_json::from_slice(right.payload.as_bytes()).unwrap();
+
+        (left.topic == right.topic) && (left_msg == right_msg)
+    } else {
+        left.topic == right.topic
+    }
```

Comment:
Instead of unwrapping the errors, one can convert these to options. This would keep the code simple.

```
    let left_msg: serde_json::Value = serde_json::from_slice(left.payload.as_bytes()).ok();
    let right_msg: serde_json::Value = serde_json::from_slice(right.payload.as_bytes()).ok();
    (left.topic == right.topic) && (left_msg == right_msg)
```

Reply:
Updated,